src/main/java/de/tum/in/www1/artemis/web/websocket/repository/RepositoryWebsocketService.java

```diff
@@ -69,53 +72,56 @@ private boolean userHasPermissions(Participation participation, Principal princi
         return true;
     }
 
+
     /**
-     * Path used for processing and persisting file updates.
-     * Sends a success message if the file changes could be saved, error messages if the process fails
-     * when the git repository can't be reached or the file can't be found.
-     * To store the update it is necessary that the user has the necessary permissions to make changes in the repository.
-     *
-     * @param participationId Participation ID
-     * @param submission
-     * @param principal
-     * @return
+     * Retrieve the file from repository and update its content with the submission's content.
+     * Throws exceptions if the user doesn't have permissions, the file can't be retrieved or it can't be updated.
+     * @param participationId id of participation to which the file belongs
+     * @param submission information about file update
+     * @param principal used to check if the user can update the file
+     * @throws InterruptedException
+     * @throws IOException
+     * @throws NoPermissionException
      */
-    @MessageMapping("/topic/repository/{participationId}/file")
-    public void updateFile(@DestinationVariable Long participationId, @Payload FileSubmission submission, Principal principal) {
+    private void fetchAndUpdateFile(Long participationId, FileSubmission submission, Principal principal) throws InterruptedException, IOException, NoPermissionException {
         Participation participation = participationService.findOne(participationId);
         if (checkParticipation(participation, principal)) {
```

Comment:
Please move the database call, the permission check and the repo checkout (which includes a pull operation) out of the for loop so that it is only executed once even if 10 files are submitted. Just move this code to the method `updateFiles` and pass the repository as variable in the method

Reply:
Good idea! I changed it accordingly.